internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -904,12 +978,145 @@ func (_ *Server) processInstallationAppEvent(
 		WithProviderClass(db.ProviderClassGithubApp).
 		WithPayload(payloadBytes)
 
-	res := &processingResult{
-		topic: installations.ProviderInstallationTopic,
-		iiw:   iiw,
+	return []*processingResult{
+		{
+			topic: installations.ProviderInstallationTopic,
+			iiw:   iiw,
+		},
+	}, nil
+}
+
+// processInstallationRepositoriesAppEvent processes events related to
+// changes to the list of repositories that the app can access.
+func (s *Server) processInstallationRepositoriesAppEvent(
+	ctx context.Context,
+	payload []byte,
+) ([]*processingResult, error) {
+	var event *installationRepositoriesEvent
+	if err := json.Unmarshal(payload, &event); err != nil {
+		return nil, err
+	}
+
+	// Check fields mandatory for processing the event
+	if event.GetAction() == "" {
+		return nil, errors.New("invalid event: action is nil")
+	}
+	if event.GetInstallation() == nil {
+		return nil, errors.New("invalid event: installation is nil")
+	}
+	if event.GetInstallation().GetID() == 0 {
+		return nil, errors.New("invalid installation: id is 0")
+	}
+
+	installationID := event.GetInstallation().GetID()
+	installation, err := s.store.GetInstallationIDByAppID(ctx, installationID)
+	if errors.Is(err, sql.ErrNoRows) {
+		return nil, fmt.Errorf("no installation found for id %d", installationID)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("could not determine provider id: %v", err)
+	}
+	if !installation.ProviderID.Valid {
+		return nil, errors.New("invalid provider id")
+	}
+	if !installation.ProjectID.Valid {
+		return nil, errors.New("invalid project id")
+	}
+
+	results := make([]*processingResult, 0, len(event.GetRepositoriesAdded())+len(event.GetRepositoriesRemoved()))
+	for _, repo := range event.GetRepositoriesAdded() {
+		// caveat: we're accessing the database once for every
+		// repository, which might be inefficient at scale.
+		res, err := s.repositoryAdded(
+			ctx,
+			repo,
+			event.GetAction(),
+			events.TopicQueueReconcileEntityAdd,
```

Comment:
can the topic ever be something else (asking if it needs to be a parameter)

Reply:
You're right, this can only be a `events.TopicQueueReconcileEntityAdd`.
Initially the routine was just one, this is a leftover.